niimpy/EDA/test_EDA_punchcard.py

```diff
@@ -1,113 +1,47 @@
-                                         resample = 'D',
-                                         normalize = False,
-                                         timerange = ('20190125','20190701'))
-                                                  
-        expected_error_msg = "title is not a string."
-        # Check if the raised ValueError contains the correct message
-        assert exc_info.match(expected_error_msg)
-        
-        # Store information about raised ValueError in exc_info
-        with pytest.raises(AssertionError) as exc_info:
-            EDA_punchcard.punchcard_plot(df,
-                                         user_list = ['user_1'],
-                                         columns = ['col_1'],
-                                         title = 'title',
-                                         resample = 1,
-                                         normalize = False,
-                                         timerange = ('20190125','20190701'))
-                                                  
-        expected_error_msg = "resample is not a string."
-        # Check if the raised ValueError contains the correct message
-        assert exc_info.match(expected_error_msg)
-        
-        # Store information about raised ValueError in exc_info
-        with pytest.raises(AssertionError) as exc_info:
-            EDA_punchcard.punchcard_plot(df,
-                                         user_list = ['user_1'],
-                                         columns = ['col_1'],
-                                         title = 'title',
-                                         resample = 'D',
-                                         normalize = 'False',
-                                         timerange = ('20190125','20190701'))
-                                                  
-        expected_error_msg = "normalize is not a boolean."
-        # Check if the raised ValueError contains the correct message
-        assert exc_info.match(expected_error_msg)
-        
-        # Store information about raised ValueError in exc_info
-        with pytest.raises(AssertionError) as exc_info:
-            EDA_punchcard.punchcard_plot(df,
-                                         user_list = ['user_1'],
-                                         columns = ['col_1'],
-                                         title = 'title',
-                                         resample = 'D',
-                                         normalize = False,
-                                         timerange = ['20190125','20190701'])
-                                                  
-        expected_error_msg = "timerange is not a boolean or tuple."
-        # Check if the raised ValueError contains the correct message
-        assert exc_info.match(expected_error_msg)
+    fig = EDA_punchcard.punchcard_plot(df,user_list,columns,title, resample,normalize,timerange)
+    assert (type(fig) == plotly.graph_objs._figure.Figure)    
```

Comment:
you can check if the correct time range is displayed here by calling `figure.data[0].x`. It will return the xarray that is shown to users.

Reply:
Added a test for testing a datetime